waypoint: 0.2.0 -> 0.2.1

06kellyjac opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

- General cleanup
- Restructured so make bin succeeds
- No need to destribute waypoint-entrypoint

Last time we had issues with the resulting binary not being able to successfully inject the `waypoint-entrypoint` without adding extra static flags.
(#103754)

![image](https://user-images.githubusercontent.com/9866621/107119902-671e2c80-6882-11eb-841e-9a00b7e3b382.png)

Using a more modern and maintained version of go-bindata resolved this issue.

@winpat if you could double check this works that'd be great but `waypoint build` and `waypoint up` with a the example project and no actual deploy target worked fine.


This version of go-bindata is also used by influxdb2.
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/servers/nosql/influxdb2/default.nix#L77-L91

Potentially it should replace the current version we're using for go-bindata. It's used by homebrew.

If that's desireable I suggest that be done in a separate PR as there are plenty of packages that use it and it'll probably need to go into staging.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS (x86_64)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

06kellyjac commented:
Result of `nixpkgs-review pr 112148` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>waypoint</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/applications/networking/cluster/waypoint/default.nix:
```diff
@@ -1,46 +1,59 @@
   '';
 
+  dontPatchELF = true;
+  dontPatchShebangs = true;
```
Why do we want this?

06kellyjac replied:
Because the result is static

06kellyjac replied:
With:
```
λ nix-build -A waypoint
this derivation will be built:
  /nix/store/mblanxb2j1lgcwx6z2yzamlj3irn5ynp-waypoint-0.2.1.drv
...
post-installation fixup
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/8n97lldddgl4aj3gvgh1h552px1vaii4-waypoint-0.2.1/bin
checking for references to /build/ in /nix/store/8n97lldddgl4aj3gvgh1h552px1vaii4-waypoint-0.2.1...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/8n97lldddgl4aj3gvgh1h552px1vaii4-waypoint-0.2.1
```

Without:
```
λ nix-build -A waypoint
this derivation will be built:
  /nix/store/7099x1dmcb27x6pij6wchd9zwa971igh-waypoint-0.2.1.drv
...
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1
shrinking /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1/bin/waypoint
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1/bin
patching script interpreter paths in /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1
checking for references to /build/ in /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1

Without + CGO_ENABLED=1 on the final waypoint bin
(as a note this isn't desirable because the final bin is still 122MB (same as static) but now it also requires glibc)
```
λ nix-build -A waypoint
this derivation will be built:
  /nix/store/4ka4jf15ir9rk0nk4csd0ab2pcxdzb00-waypoint-0.2.1.drv
...
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1
shrinking /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1/bin/waypoint
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1/bin
patching script interpreter paths in /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1
checking for references to /build/ in /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1...
/nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1
```

06kellyjac replied:
As a note I'm getting this error with all builds:

```
running tests
/nix/store/h50ypqigilk8cv8a2gvwr2x2gl6pfq8v-stdenv-linux/setup: line 1310: getGoDirs: command not found
```

I don't think I'm doing anything special to break it

06kellyjac commented:
I've made a PR for the maintained version of go-bindata #112164

If it can be merged easily into master that one should go in before this one (and I can clean this copy up)
Or if it has to go into staging then I think this can merge into master until it makes its way through

winpat commented:
Not sure why, but my local build fails with:

```
...
patching sources
configuring
building
vendor folder exists, please set 'vendorSha256 = null;' in your expression
builder for '/nix/store/yw81b341hz0hdxf9j8n9bhnjvmx1x9rl-waypoint-0.2.1-go-modules.drv' failed with exit code 10
...
```

Any idea why this did not happen in your case @06kellyjac ?

06kellyjac commented:
So there is a vendor folder with just a git submodule.
I think deleteVendor = true; allowed me to get the sha for the new vendor
Then when I removed it, because the sha was in my nix-store it just used that before checking if there was the pre-existing vendor

SuperSandro2000 commented on pkgs/applications/networking/cluster/waypoint/default.nix:
```diff
@@ -1,46 +1,60 @@
+let
+  # nixpkgs go-bindata causes issues with the waypoint-entrypoint
+  go-bindata = buildGoPackage rec {
+    pname = "go-bindata";
```
I just merged the PR that adds go-bindata globally. Can we use that here?

06kellyjac replied:
Yup its the same version. Ill rebase

06kellyjac replied:
rebased

06kellyjac committed:
waypoint: 0.2.0 -> 0.2.1

- General cleanup
- Restructured so make bin succeeds
- No need to destribute waypoint-entrypoint

06kellyjac commented:
Result of `nixpkgs-review pr 112148` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>waypoint</li>
  </ul>
</details>
